sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1619,8 +1626,31 @@ def get_dom_polling_from_config_db(self, lport):
 
         return dom_polling
 
+    """
+    Checks if the port is in CMIS terminal state
+    This API assumes CMIS_STATE_UNKNOWN as a transitional state since any CMIS supported platform will
+    eventually reach to a CMIS terminal state (CMIS_TERMINAL_STATES) irrespective of the transciver type
+    Returns:
+        True if the port is in CMIS initialization process, otherwise False
+    """
+    def is_port_in_cmis_terminal_state(self, logical_port_name):
```

Comment:
@mihirpat1 instead of passing cmismgr to dom thread, can we post the CMIS state to `TRANSCEIVER_STATUS` table and use this API to fetch it from that table? this will also help in cases to know the current SW state machine state

Reply:
I have now replaced using cmismgr with updating `TRANSCEIVER_STATUS ` table to communicate the `cmis_state` between the threads.